packages/material-ui/src/LinearProgress/LinearProgress.d.ts

```diff
@@ -7,8 +7,13 @@ export interface LinearProgressProps
   value?: number;
   valueBuffer?: number;
   variant?: 'determinate' | 'indeterminate' | 'buffer' | 'query';
+  styles?: { [P in LinearProgressStylesKeys]?: React.CSSProperties };
```

Comment:
Just as an FYI - this line is equivalent to this:
```ts
styles?: {bar1?: React.CSSProperties; bar2?: React.CSSProperties; };
```

Reply:
I would suggest going the other direction,

```ts
styles?: Partial<Record<LinearProgressStylesKeys, React.CSSProperties>>
```

🤷‍♂️

Reply:
@pelotom I didn't know about `Record`. Good suggestion.